src/gui/connectionWidget.cpp

```diff
@@ -60,9 +64,19 @@ void ConnectionWidget::requestNewFreq(int index) {
     }
 }
 
+void ConnectionWidget::requestNewUnit(int index) {
+    if (index >= 0 && index < ui->boxUnit->count()) {
+        UnitValue unit = UnitValue(ui->boxUnit->currentData().toInt());
+        if (communication != nullptr) {
+            communication->setNewUnit(unit);
+        }
+    }
+}
+
 void ConnectionWidget::updateWidget(Sample readings) {
     updateBatteryIcon(readings.batteryPercent);
     updateCurrentFrequency(readings.frequency);
+    updateUnit(readings.unitValue);
```

Comment:
Maybe add `TODO: investigate performance`, if this triggers a rerender, or sends a host of events on every sample that could be bad.

Reply:
Tracked in #90